web-authentication-ui/src/main/java/com/okta/webauthenticationui/ForegroundViewModel.kt

```diff
@@ -17,48 +17,58 @@ package com.okta.webauthenticationui
 
 import android.app.Activity
 import android.net.Uri
+import android.os.Parcelable
 import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import kotlinx.coroutines.launch
-import okhttp3.HttpUrl
+import kotlinx.parcelize.Parcelize
+import okhttp3.HttpUrl.Companion.toHttpUrl
 
-internal class ForegroundViewModel : ViewModel() {
+internal class ForegroundViewModel(private val savedStateHandle: SavedStateHandle) : ViewModel() {
```

Comment:
This is the fix. savedStateHandle survives activity onDestroy() when system kills the activity. This replaces _stateLiveData.